env-code-challenges/bin/Activate.ps1

```diff
@@ -0,0 +1,247 @@
+<#

```

Comment:
`venv` is the conventional name for the virtual environment.  `python -m venv venv` <---

Since this environment isn't name conventioanlly, it isn't in the `.gitignore` here https://github.com/nxnode/Code-Challenges/blob/main/.gitignore#L108 

To fix this,
1) delete this directory locally
2) add and commit the changes and push to this branch on github
3) either add this name to the .gitignore or use the conventional name.
4) _if_ you're opting to keep the name like it is git add, commit, and push the changes to the .gitignore to this branch.

Reply:
I think it should be good now after doing the steps
I added it to the .gitignore and I am just going to use the unconventional name for now - I feel like I probably did everything in the wrong order but I deleted it locally, added env-code-challenges/ to the .gitignore and then after commit and push,

I can then just 'python -m venv env-code-challenges' again and it should not be included in the next branch/commit, right?

Reply:
Is it okay to just checkout a new branch and work on something or is it best to wait for the merge?

Reply:
> Is it okay to just checkout a new branch

Yes, though you might get yourself into a situation were there are merge conflicts, which is okay and normal. I'd just have to show you haw to handle them.

Reply:
> I probably did everything in the wrong order

It appears to have worked.  The order matters to some extent but in this case seems to not have.  If a file isn't in the .gitignore it is tracked.  If it's added to github before being added to the ignore and commited, deleting it locally before adding it to the ignore might not remove it from github.   This might only be an issue when a file is merged and no longer on a feature branch 🤷‍♂️ 

This commit isn't dangerous, but imagine you had commited a file with passwords or API tokens in it.  It isn't technically removed from Github.  It'll be in [that commit](https://github.com/nxnode/Code-Challenges/commit/e30cd50a8ed28bb7780039bf22df71d824db3c75) until the repo is deleted, and maybe even then it would exist just you would no longer have access.  To prevent casual browsing/crawlers from finding the commit you could `sqaush merge` this commmit, it basically puts all the commits into a single on.  Then the troublesome commit is removed from the history.  It still exists by direct url calls, but wouldn't show up anywhere.  Best is just to always be very careful with what you commit.  `git add .` is dangerous in this way.  `git add <path to file>` is safer.  Also `git status` can show what is being tracked.